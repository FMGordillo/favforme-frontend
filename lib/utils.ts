/**
 * @default 0
 */
export const toPascalCase = (string: string | undefined): string =>
  string
    ? string.replace(/(\w)(\w*)/g, function (_g0, g1, g2) {
        return g1.toUpperCase() + g2.toLowerCase();
      })
    : "";
