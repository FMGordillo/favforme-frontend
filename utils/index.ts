// export const isNotProd = process.env.NEXT_PUBLIC_ENVIRONMENT !== "production";
export const isNotProd = true;
