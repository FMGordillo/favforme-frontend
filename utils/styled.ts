import { DefaultTheme } from "styled-components";

export type Direction = "left" | "center" | "right";

// TODO: Use this from styled.ts
export type Color = "primary" | "secondary" | "gray" | "white" | "black";
export type Variant = "light" | "main" | "dark";
export type Spacing = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
export type Breakpoint = "xs" | "sm" | "md" | "lg" | "xl";

export const calculateBreakpoint = (key: Breakpoint): string => {
  switch (key) {
    case "xs":
      return "0px";
    case "sm":
      return "600px";
    case "md":
      return "960px";
    case "lg":
      return "1280px";
    case "xl":
      return "1920px";
  }
};

const defaultTheme = {
  spacing: (digit: Spacing): string => `${digit * 4}`,
  breakpoints: {
    down: (key: Breakpoint): string =>
      `@media (max-width: ${calculateBreakpoint(key)})`,
    // @deprecated
    up: (key: Breakpoint): string =>
      `@media (min-width: ${calculateBreakpoint(key)})`,
  },
};

export const lightTheme: DefaultTheme = {
  mode: "light",
  palette: {
    primary: { main: "#4f6be6", light: "#9BAAF6", dark: "#27358d" },
    secondary: {
      light: "#FFBCC4",
      main: "#f38d9f",
      dark: "#a21d44",
    },
    gray: { light: "#f6f6f6", main: "#B2B2B2", dark: "#797979" },
    black: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    white: {
      light: "#FFF",
      main: "#FFF",
      dark: "#FFF",
    },
    error: "#b50404",
  },
  ...defaultTheme,
};

export const darkTheme: DefaultTheme = {
  mode: "dark",
  palette: {
    // primary: { main: "#5573f5", light: "#9BAAF6", dark: "#27358d" },
    primary: { main: "#27358d", light: "#27358d", dark: "#27358d" },
    secondary: {
      light: "#FFBCC4",
      main: "#f38d9f",
      dark: "#a21d44",
    },
    // gray: { light: "#f6f6f6", main: "#B2B2B2", dark: "#9b9b9b" },
    gray: { light: "#9b9b9b", main: "#9b9b9b", dark: "#2f2f2f" },
    black: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    white: {
      light: "#000",
      main: "#000",
      dark: "#000",
    },
    error: "#b50404",
  },
  ...defaultTheme,
};
