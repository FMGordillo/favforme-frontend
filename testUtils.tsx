import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "./utils/styled";
// import { ThemeProvider } from "my-ui-lib"
// import { TranslationProvider } from "my-i18n-lib"
// import defaultStrings from "i18n/en-x-default"

const Providers = ({ children }) => {
  // return children;
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
