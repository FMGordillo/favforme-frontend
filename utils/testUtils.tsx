import { render } from "@testing-library/react";
import { ReactChild } from "react";
import { ThemeProvider } from "styled-components";
import { SWRConfig } from "swr";
import { fetcher } from "../lib/queries";
import { lightTheme } from "./styled";

const Providers = ({ children }: { children: ReactChild }) => {
  return (
    <SWRConfig
      value={{
        refreshInterval: 10000,
        fetcher,
      }}
    >
      <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
    </SWRConfig>
  );
};

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const customRender = (ui: JSX.Element, options = {}) =>
  // @ts-ignore
  render(ui, { wrapper: Providers, ...options });

import "@testing-library/jest-dom";
export * from "@testing-library/react";
export { customRender as render };
