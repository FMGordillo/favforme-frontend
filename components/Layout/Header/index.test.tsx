import { Header } from "./";
import { render } from "@/utils/testUtils";

// TODO: improve tests here!
describe("[Componentes] Header", () => {
  it("renders correctly", async () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
