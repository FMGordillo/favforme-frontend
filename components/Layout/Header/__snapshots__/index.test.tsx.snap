// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Componentes] Header renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  left: 0;
  right: 0;
  z-index: 9;
  max-height: 250px;
  border-radius: 0 0 100% 100%/0 0 25% 25%;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c1 {
  display: none;
  position: fixed;
  top: 0;
  z-index: 9;
  height: 4em;
  width: 100vw;
  padding: 1em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: dosis,sans-serif;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 500;
  text-align: center;
  max-width: 200px;
  display: block;
}

.c3 > a > div {
  max-width: 150px;
}

.c3 a {
  color: white;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: dosis,sans-serif;
  text-transform: uppercase;
  font-size: 15px;
  font-weight: 500;
  text-align: center;
  max-width: 200px;
  display: block;
}

.c5 > a > div {
  max-width: 150px;
}

.c5 a {
  color: #f38d9f;
}

.c4 {
  font-weight: 500;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-bottom: 1px solid transparent;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.c4:hover {
  font-weight: 700;
  color: #f38d9f;
}

@media (max-width:600px) {
  .c0 {
    color: red;
  }
}

@media (max-width:600px) {
  .c1 {
    display: block;
  }
}

@media (max-width:600px) {
  .c2 {
    -webkit-transform: translateY(-100%);
    -ms-transform: translateY(-100%);
    transform: translateY(-100%);
    background: #27358d;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding-top: 80px;
    margin: 0;
    -webkit-transition: -webkit-transform 250ms ease-in;
    -webkit-transition: transform 250ms ease-in;
    transition: transform 250ms ease-in;
    position: fixed;
    width: 100vw;
  }

  .c2 > :nth-child(3) {
    display: none;
  }

  .c2 > :nth-child(3) {
    display: none;
  }

  .c2 > * {
    margin: 1em;
  }

  .c2 > :last-child {
    margin-bottom: 2em;
  }
}

@media (max-width:600px) {
  .c3 {
    display: block;
  }
}

@media (max-width:600px) {
  .c5 {
    display: block;
  }
}

@media (max-width:960px) {

}

@media (max-width:960px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {

}

<nav
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 60px; height: 60px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <ul
      class="c2"
    >
      <li
        class="c3"
      >
        <a
          class="c4"
        >
          FavForMe
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c4"
        >
          Acciones
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c4"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjcwIiBoZWlnaHQ9IjIxNCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="FavForMe logo"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </a>
      </li>
      <li
        class="c3"
      >
        <a
          class="c4"
        >
          Empresas
        </a>
      </li>
      <li
        class="c5"
      >
        <a
          class="c4"
        >
          Contacto
        </a>
      </li>
    </ul>
  </nav>
  .c0 {
  background-size: cover;
  background-position: center;
  background-image: url("/images/plato_de_comida.webp");
  height: 400px;
  width: 100vw;
  overflow: hidden;
  display: table-cell;
  vertical-align: bottom;
  padding-bottom: 48px;
  -webkit-transition: height 300ms;
  transition: height 300ms;
}

.c1 {
  color: white;
  -webkit-transform: translateY(-64px);
  -ms-transform: translateY(-64px);
  transform: translateY(-64px);
  padding: 0;
  -webkit-transition: -webkit-transform 300ms;
  -webkit-transition: transform 300ms;
  transition: transform 300ms;
}

.c2 {
  margin: 0;
  font-size: 50px;
  text-overflow: ellipsis;
  overflow: hidden;
  width: inherit;
  text-align: center;
}

.c2 mark {
  color: white;
  background-color: #4f6be6;
}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:960px) {
  .c0 {
    height: 300px;
  }
}

@media (max-width:960px) {
  .c1 {
    -webkit-transform: translateY(0px);
    -ms-transform: translateY(0px);
    transform: translateY(0px);
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0;
  }
}

@media (max-width:960px) {
  .c2 {
    width: inherit;
  }
}

<header
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        <span>
          Hay muchas personas
          <br />
          necesitando de tu
          <br />
          buena acci√≥n
        </span>
      </h1>
    </div>
  </header>
</DocumentFragment>
`;
